src/hotspot/share/gc/shared/space.cpp

```diff
@@ -126,7 +126,9 @@ HeapWord* ContiguousSpace::forward(oop q, size_t size,
     cp->space->set_compaction_top(compact_top);
     cp->space = cp->space->next_compaction_space();
     if (cp->space == nullptr) {
-      cp->gen = GenCollectedHeap::heap()->young_gen();
+#if INCLUDE_SERIALGC
+      cp->gen = SerialHeap::heap()->young_gen();
+#endif // INCLUDE_SERIALGC
```

Comment:
This doesn't look right. `INCLUDE_SERIALGC` and its counterparts in other GCs are to support release builds without certain GCs. IOW, when this is `false`, it should still build. I believe https://github.com/openjdk/jdk/pull/16842 should make this part of change obsolete.